lib/fizzy/parser_expr.cpp

```diff
@@ -170,6 +170,56 @@ parser_result<Code> parse_expr(const uint8_t* pos)
             break;
         }
 
+        case Instr::if_:
+        {
+            const uint8_t type{*pos};
+            uint8_t arity;
+            if (type == BlockTypeEmpty)
+            {
+                arity = 0;
+                ++pos;
+            }
+            else
+            {
+                // Will throw in case of incorrect type.
+                std::tie(std::ignore, pos) = parser<ValType>{}(pos);
+                arity = 1;
+            }
+
+            code.immediates.push_back(arity);
+
+            label_positions.push_back({Instr::if_, code.immediates.size()});
+
+            // Placeholders for immediate values, filled at the matching end and else instructions.
+            push(code.immediates, uint32_t{0});  // Diff to the end instruction.
+            push(code.immediates, uint32_t{0});  // Diff for the immediates
+
+            push(code.immediates, uint32_t{0});  // Diff to the else instruction
+            push(code.immediates, uint32_t{0});  // Diff for the immediates.
+
+            break;
+        }
+
+        case Instr::else_:
+        {
+            if (label_positions.empty())
+                throw parser_error{"unexpected else instruction"};
```

Comment:
Can you also add a test case triggering this?
```
(func
  else  
)
```

Reply:
Done.